### DISCUSSION TITLE
how to control timestamp conversion format?

### DISCUSSION THREAD

**Discussion by avoidik** [mikefarah/yq#1699] [General]
Hello,

I am trying to convert a yaml file to json file. Is there any global option to NOT convert timestamp properties according with RFC 3339? I would like to keep them as is.

```
-  "Version": "2012-10-17T00:00:00Z"
+  "Version": "2012-10-17"
```

```bash
yq eval -p yaml -o json -P input.yml > output.json
```

Regards

**Comment by avoidik** [comment]
if I run this

```bash
yq --null-input -p yaml -o json '.Version = "2012-10-17" | .Version |= format_datetime("2006-01-02")'
```

it is still

```
{
  "Version": "2012-10-17T00:00:00Z"
}
```

same with

```bash
yq --null-input -p yaml -o json '.Version = "2012-10-17" | .Version |= with_dtf("2006-01-02"; format_datetime("2006-01-02"))'
```

why?

**Comment by mikefarah** [comment]
That is odd. Looking into it a little more, I can see what causes it (and I need to think about it a little more). When the `format_datetime` operator is applied to 'version', it changes the property to a '!!timestamp' type. The issue is, when it goes through the JSON encoder, it sees it's a timestamp and reformats it.

As a work-around till that's fixed, you can do the following:

a) don't use the formatter
```
yq --null-input -p yaml -o json '.Version = "2012-10-17"'
```

or b) use the formatter, and force it back to a string
```
yq --null-input -p yaml -o json 'with(.Version;  .= "2012-10-17" | . |= format_datetime("2006-01-02") | . tag="!!str")'
```

**Reply by avoidik** [reply]
much appreciated for coming back with your comment and workaround, and also confirming an edge-case issue, I think I will wait for the fix to arrive.

**Reply by avoidik** [reply]
@mikefarah should I create an issue instead, what do you think?
